### PR TITLE
Add scripts and docs for TREC Fair 2022

### PR DESCRIPTION
**Pull request by alvind1**
Add scripts and documentation guide for reproducing BM25, BM25+RM3 baselines for the TREC 2022 Fair Rankings Track

### CODE REVIEW

File: docs/experiments-trec2022-fairness-ranking.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+# Pyserini: BM25 and RM3 Baselines for TREC 2022 Fair Ranking Track
+
+This guide contains instructions for running BM25 and RM3 baselines on the [TREC 2022 Fair Ranking Track](https://fair-trec.github.io/)
```
Comment (lintool) [inline docs/experiments-trec2022-fairness-ranking.md:3]:
```suggestion
This guide contains instructions for running BM25 and RM3 baselines on the [TREC 2022 Fair Ranking Track](https://fair-trec.github.io/).
```

File: docs/experiments-trec2022-fairness-ranking.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+# Data Prep
+
+This guide requires the [development installation](https://github.com/castorini/pyserini/blob/master/docs/installation.md#development-installation) for additional resource not shipped with the Python module.
+
+First, we need to download the Fair Ranking training dataset and the topics and queries.
+
+TREC 2022 Fair Ranking provides three different formats: plain, text, html
```
Comment (lintool) [inline docs/experiments-trec2022-fairness-ranking.md:11]:
```suggestion
TREC 2022 Fair Ranking provides three different formats: plain, text, html.
```

File: docs/experiments-trec2022-fairness-ranking.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+In this example, only plain (7.1GB) will be used, but getting the baselines for the other formats is a nearly identical process.  
+The text (16GB) and html (63GB) are significantly bigger than the plain collection so the following commands will also take longer for those collections.
+
+```
```
Comment (lintool) [inline docs/experiments-trec2022-fairness-ranking.md:16]:
```suggestion
```bash
```

Do this for all code blocks also?

File: docs/experiments-trec2022-fairness-ranking.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  --index indexes/plain \
+  --generator DefaultLuceneDocumentGenerator \
+  --threads 9 \
+  --storePositions --storeDocvectors --storeRaw
+```
+This takes about 1h. There should be 6,475,537 documents indexed.
+
+# Performing Retrieval on Training Queries
+
+Using BM25
```
Comment (lintool) [inline docs/experiments-trec2022-fairness-ranking.md:63]:
```suggestion
Using BM25:
```

Same below?

File: docs/experiments-trec2022-fairness-ranking.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  --topics tools/topics-and-qrels/trec_fair_2022_queries.tsv \
+  --output runs/run.plain.bm25.rm3.txt \
+  --bm25 \
+  --rm3 \
+  --hits 500
+```
+
+# Evaluation
+
+To evaluate, we first need to convert the given relevant documents into qrels format
```
Comment (lintool) [inline docs/experiments-trec2022-fairness-ranking.md:86]: Just be consistent on ending... sometimes you have colon, sometimes comma, sometimes nothing...

File: docs/experiments-trec2022-fairness-ranking.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+```
+python -m pyserini.search.lucene \
+  --index indexes/plain \
+  --topics tools/topics-and-qrels/trec_fair_2022_queries.tsv \
+  --output runs/run.plain.bm25.txt \
+  --bm25 \
+  --hits 500
+```
+
+Using BM25+RM3
```
Comment (lintool) [inline docs/experiments-trec2022-fairness-ranking.md:73]: Can we add Rocchio also?

Just change `--bm25` to `--rocchio`. Everything else should "just work".

**Review by lintool — CHANGES_REQUESTED** [review]

File: docs/experiments-trec2022-fairness-ranking.md (hunk 6 above)
Comment (alvind1) [reply docs/experiments-trec2022-fairness-ranking.md:73]: So there should be four baselines? bm25, bm25+rm3, rocchio, rocchio + rm3

Comment (lintool) [reply docs/experiments-trec2022-fairness-ranking.md:73]: No, Rocchio is mutually exclusive w/ BM25.
So BM25 + {RM3, Rocchio_

Comment (alvind1) [reply docs/experiments-trec2022-fairness-ranking.md:73]: Oh okay, just to confirm, three baselines: bm25, bm25+rm3, bm25+rocchio?

Comment (lintool) [reply docs/experiments-trec2022-fairness-ranking.md:73]: correct.

**Review by lintool — APPROVED** [review]

**Pull request merged**